### DISCUSSION TITLE
static lifetime for `CloseFrame` in `ws::Message::Close`

### DISCUSSION THREAD

**Discussion by sukhmel** [tokio-rs/axum#2065] [Q&A] [answered]
Hello,

I've been wondering why `ws::Message` has its variant defined as `Close(Option<CloseFrame<'static>>)`.
Is it because of some limitation of ws implementation, because of the hard-to-express lifetime, because you did not want `ws::Message` to have mostly useless lifetime specifier, or for some other reason?

It's usually not a problem but sometimes for better diagnosis, it would be simpler to be able to pass dynamic data. If that's something that may/should be fixed, I may give it a try, but if that's not desired, that's also fine

**Comment by davidpdrsn** [comment, marked as answer]
You can pass dynamic data. The lifetime is used in a Cow<‘static, str> which means either a &’static str or a dynamic String. It doesn’t require a &’static str.

**Comment by sukhmel** [comment]
Oh, that's just me failing to use `Cow` property then, thank you for a
clarification.

чт, 29 июн. 2023 г., 19:54 David Pedersen ***@***.***>:
